traits/trait_types.py

```diff
@@ -1829,192 +1960,180 @@ def init_fast_validator(self, *args):
-            if low is not None:
-                vtype = type(low)
-            elif high is not None:
-                vtype = type(high)
-            else:
-                vtype = lambda x: x
-
-        return vtype(value)
+        value = self._validate(object, name, value)
 
-    def _set_value(self, object, name, value):
-        """ Sets the specified value as the value of the dynamic range.
-        """
         cname = "_traits_cache_" + name
         old = object.__dict__.get(cname, Undefined)
         if old is Undefined:
-            old = eval(self._value)
+            # Again, for consistency with the non-clipped case we should really
+            # be validating here. See the comment in _get.
+            old = self._get_default_value(object)
         object.__dict__[cname] = value
         if value != old:
             object.trait_property_changed(name, old, value)
 
     def full_info(self, object, name, value):
         """ Returns a description of the trait.
         """
-        if self._vtype is not Undefined:
-            low = eval(self._low)
-            high = eval(self._high)
-            low, high = (
-                self._typed_value(low, low, high),
-                self._typed_value(high, low, high),
-            )
-        else:
-            low = self._low
-            high = self._high
+        low, high = self._get_bounds(object)
+
+        value_info = self._value_trait.full_info(object, name, value)
 
         if low is None:
             if high is None:
-                return self._type_desc
+                return value_info
 
             return "%s <%s %s" % (
-                self._type_desc,
-                "="[self._exclude_high :],
+                value_info,
+                "="[self._exclude_high:],
                 high,
             )
 
         elif high is None:
             return "%s >%s %s" % (
-                self._type_desc,
-                "="[self._exclude_low :],
+                value_info,
+                "="[self._exclude_low:],
                 low,
             )
 
         return "%s <%s %s <%s %s" % (
             low,
-            "="[self._exclude_low :],
-            self._type_desc,
-            "="[self._exclude_high :],
+            "="[self._exclude_low:],
+            value_info,
+            "="[self._exclude_high:],
             high,
         )
 
+    def inner_traits(self):
+        """ Returns the *inner trait* (or traits) for this trait.
+        """
+        return (self._value_trait,)
+
     def create_editor(self):
         """ Returns the default UI editor for the trait.
         """
+        from traitsui.api import RangeEditor
+
         # fixme: Needs to support a dynamic range editor.
 
         auto_set = self.auto_set
         if auto_set is None:
             auto_set = True
 
-        from traitsui.api import RangeEditor
+        if self._low_name is None:
+            low_name = ""
+        else:
+            low_name = "object." + self._low_name
```

Comment:
This is not right, but it's not the `Range` trait's fault - it's not being given the context that it needs from TraitsUI. See #596

Reply:
Comment (with issue reference) added to the code; thanks for opening the issue.